File: AVc/app/src/main/java/com/hfad/avc/ui/main_avtivity/MainPresenter.java
package com.lastaurus.automatic_congratulations.ui.main_avtivity;

import com.github.terrakok.cicerone.Router;
import com.lastaurus.automatic_congratulations.cicerone.Screens;
import com.lastaurus.automatic_congratulations.dagger.ComponentManager;

import javax.inject.Inject;

import moxy.InjectViewState;
import moxy.MvpPresenter;


@InjectViewState
public class MainPresenter extends MvpPresenter<IMainViewModel> {

   @Inject
   Router router;

   public MainPresenter() {
      ComponentManager.Companion.getInstance().appComponent.inject(this);
   }

//   public void openTemplateWrite() {
//      this.router.navigateTo(Screens.changeTemplateScreen(-1));
//   }
//
//   public void openDbContact() {
//      this.router.navigateTo(Screens.conatctListScreen());
//   }
//
//   public void openTemplateList() {
//      this.router.navigateTo(Screens.templateListScreen());
//   }

   @Override
   protected void onFirstViewAttach() {
      super.onFirstViewAttach();
      this.router.navigateTo(Screens.main());
   }
}

File: app/src/main/java/com/lastaurus/automatic_congratulations/ui/main_avtivity/MainPresenter.kt
package com.lastaurus.automatic_congratulations.ui.main_avtivity

import com.github.terrakok.cicerone.Router
import com.lastaurus.automatic_congratulations.bus.RxBus
import com.lastaurus.automatic_congratulations.cicerone.Screens.main
import com.lastaurus.automatic_congratulations.dagger.ComponentManager.Companion.instance
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
import io.reactivex.rxjava3.disposables.CompositeDisposable
import io.reactivex.rxjava3.schedulers.Schedulers
import moxy.InjectViewState
import moxy.MvpPresenter
import javax.inject.Inject

@InjectViewState
class MainPresenter : MvpPresenter<IMainViewModel>() {
   @Inject
   lateinit var router: Router

   @Inject
   lateinit var rxBus: RxBus
   private val disposables = CompositeDisposable()

   init {
      instance.appComponent.inject(this)
   }

   override fun onFirstViewAttach() {
      super.onFirstViewAttach()
      router.navigateTo(main())
   }
}